### PR TITLE
api: migrate utils to ethers v6 and replace BigNumber with bigint

### PR DESCRIPTION
**Pull request by sidneyarida**
## Description

#1425 

Migration of utils functions and BigNumber to ether-v6 in carbonmark-api folder

## Related Ticket

Closes <issue-number>
Also related to <issue-number>

## Notes For QA

Specific pages, components or journeys that might be affected:
- 

Relevant preview URLs:
- 

Other notes:
-

### CODE REVIEW

**Review by biwano — APPROVED** [review]
Nice PR.
Can you just check that logging in still works in carbonmark before merging?
Tests do not cover the `verifyProfileSignature` function.

File: carbonmark-api/tsconfig.json
Comment (0xemc) [inline carbonmark-api/tsconfig.json:4]: why was this necessary?

**Comment by sidneyarida** [comment]
I tried the to login and still works

File: carbonmark-api/tsconfig.json
Comment (sidneyarida) [reply carbonmark-api/tsconfig.json:4]: It was added to support this syntax for declaring bigint
const bigIntNumber = 0**n**

**Comment by 0xemc** [comment]
@sidneyarida this is good to go once you resolve the conflicts

**Comment by sidneyarida** [comment]
Giving an error here: https://carbonmark-12kti3f42-klimadao.vercel.app/activities
Seems to got broken by this pr: https://github.com/KlimaDAO/klimadao/pull/2189 not working there as well: https://carbonmark-api-git-tufnel-standardize-api-amoun-1cb01d-klimadao.vercel.app/activities

**Pull request merged**
